include/symbol.hpp

```diff
@@ -16,18 +16,42 @@ struct SymbolEntry {
       : id{std::move(id)}, type{std::move(expr_type)} {}
 };
 
-class SymbolTable {
+struct DeclTypeEntry {
```

Comment:
How about simply `TypeEntry`? Since both the symbol and type are declared, I think we don't need to encode this in the name.